buzsaki_lab_to_nwb/original_code/convert_yuta.m

```diff
@@ -1,162 +1,301 @@
+%         'data', lfp_data(:,channel_groups{i}+1), ...
+%         'electrode_group', types.untyped.Link(...
+%             ['/general/extracellular_ephys/', [device_name, '_electrodes']],...
+%             '', eg), ...
+%         'timestamps', lfp_tt');
+%     
+%     %lfp = types.LFP('source',{fname},'ElectricalSeries',es);
+%     file.acquisition.(['shank' num2str(i) ' lfp']) = es;
+% end
+
+
+%% Cody: new version
+nshanks = length(channel_groups);
+variables = {'x', 'y', 'z', 'impudence', 'location', 'filtering', 'group', 'label'};
+tbl = cell2table(cell(0, length(variables)), 'VariableNames', variables);
+
+% Cody: can't find device information
+device = types.core.Device(...
+    'description', '', ...
+    'manufacturer', '');
+device_name = 'unknown';
+nwb.general_devices.set(device_name, device);
+device_link = types.untyped.SoftLink(['/general/devices/' device_name]);
+% ------------------
+
+for ishank = 1:nshanks
+    group_name = ['shank' num2str(ishank)];
+    nwb.general_extracellular_ephys.set(group_name, ...
+        types.core.ElectrodeGroup( ...
+            'description', ['electrode group for shank' num2str(ishank)], ...
+   	        'location', 'unknown', ... % is 'unknown' correct standard for this? documentation doesn't indicate in either MatNWB or NWB itself
+   	        'device', device_link));
+    group_object_view = types.untyped.ObjectView( ...
+       	['/general/extracellular_ephys/' group_name]);
+
+    for ielec = 1:numel(channel_groups{ishank})
+        tbl = [tbl; {NaN, NaN, NaN, NaN, 'unknown', 'unknown', ...
+            group_object_view, [group_name 'elec' num2str(ielec)]}];
+    end
+end
 
-file.acquisition.position_sensor_1 = ss;
+electrode_table = util.table2nwb(tbl, 'all electrodes');
+nwb.general_extracellular_ephys_electrodes = electrode_table;
 
-%% load LFP
-lfp_filepath = fullfile(fpath, [fname, '.eeg']);
-lfp_file = fopen(lfp_filepath, 'r');
-lfp_data = fread(lfp_file,'int16=>int16'); % takes 45 seconds 
-lfp_data = reshape(lfp_data,[],sessionInfo.nChannels);
 
 %%
-lfp_data = lfp_data(1:100,:);
-lfp_tt = (1:length(lfp_data)) * samplingRate;
-
-%% construct LFP
-
-file.general.devices = types.untyped.Group();
-file.general.extracellular_ephys = types.untyped.Group();
-device_links = {};
-for i=1:length(channel_groups)
-    device_name = ['shank', num2str(i-1)];
-    dev = types.Device('source',{[fname, '.xml']});
-    file.general.devices.(device_name) = dev;
+electrodes_object_view = types.untyped.ObjectView( ...
+    '/general/extracellular_ephys/electrodes');
+
+electrode_table_region = types.hdmf_common.DynamicTableRegion( ...
+    'table', electrodes_object_view, ...
+    'description', 'all electrodes', ...
+    'data', [0 height(tbl)-1]');
+
+ecephys_module = types.core.ProcessingModule(...
+    'description', 'extracellular electrophysiology');
+
+for ishank = 1:nshanks
+    electrical_series = types.core.ElectricalSeries( ...
+        'timestamps', lfp_tt', ...
```

Comment:
starting time and rate should be used for regularly sampled data

Reply:
Python code for the LFP series are using the appropriate method.